packages/cli/src/InternalHooks.ts

```diff
@@ -240,21 +242,33 @@ export class InternalHooks implements IInternalHooksClass {
 
 	async onWorkflowBeforeExecute(
 		executionId: string,
-		data: IWorkflowExecutionDataProcess,
+		data: IWorkflowExecutionDataProcess | IWorkflowBase,
 	): Promise<void> {
+		let payload: EventPayloadWorkflow;
+		if ('executionData' in data) {
```

Comment:
Do you think it's worth commenting what behavior happens when? If I understood correctly, we get `IWorkflowExecutionDataProcess` when running in `regular` mode and `IWorkflowBase` when running in `queue` mode but as `main` process right?

Reply:
will add comment